### PR TITLE
CEL Strings extension macros

### PR DESCRIPTION
**Pull request by alexsnaps**

### CODE REVIEW

File: src/data/cel/strings.rs [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+                        ),
+                    })
+                }
+            };
+            if base >= this.len() {
+                return Ok((-1).into());
+            }
+            match this[base..].rfind(&*arg) {
+                None => Ok((-1).into()),
+                Some(idx) => Ok(Value::UInt(idx as u64)),
```
Comment (adam-cattermole) [inline src/data/cel/strings.rs:83]: Shouldn't this be `base +` similar to `index_of`?
```suggestion
                Some(idx) => Ok(Value::UInt(base + idx as u64)),
```

Comment (alexsnaps) [reply src/data/cel/strings.rs:83]: It doesn't try... and tests won't pass (the tests are from the doc et al from the golang version)
That being said, when I read the expectations for that method, I failed to understand them

File: src/data/cel/strings.rs [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+                .take_while(|_| {
+                    if let Some(v) = end.checked_sub(1) {
+                        end = v;
+                        true
+                    } else {
+                        false
+                    }
+                })
+                .collect::<String>()
+                .into())
```
Comment (adam-cattermole) [inline src/data/cel/strings.rs:269]: Maybe I'm missing a case but couldn't we just avoid this `take_while`? although maybe this is related to what we

```suggestion
            Ok(this
                .chars()
                .skip(start)
                .take(end - start)
                .collect::<String>()
                .into())
```

File: src/data/cel/strings.rs (hunk 1 above)
Comment (adam-cattermole) [reply src/data/cel/strings.rs:83]: Sure, thanks for [this link](https://github.com/google/cel-go/blob/master/ext/strings.go), I see this is expected

**Review by adam-cattermole — APPROVED** [review]
Has a merge conflict that needs resolving but the rest lgtm

**Pull request merged**